NetworkingServiceKit/Classes/Networking/APIConfigurations.swift

```diff
@@ -65,4 +60,15 @@ public protocol APIConfigurationType {
         
         return configuration.defaultConfiguration
     }
+    
+    @objc public static func current(bundleId: String = Bundle.main.appBundleIdentifier) -> APIConfiguration {
+        guard let configurationType = APIConfiguration.apiConfigurationType,
+              let authType = APIConfiguration.authConfigurationType else {
+            //TODO: Patrick, why is this error being "erased" with a print? how severe would this error be?
+            print("Error: ServiceLocator couldn't find the current APIConfiguration, make sure to define your own types for APIConfiguration.apiConfigurationType and APIConfiguration.authConfigurationType")
```

Comment:
cc @patrickomalley

Reply:
I don't think logging an improper configuration is uncommon, since the error message is targeted toward the developer rather than the app user where an alert would make sense. Making it an optional and returning `nil` might make the failure clearer but it would make the overall interface a little clunkier. It may make sense to `fatalError` here since we probably want to make it very obvious if this hasn't been set up.

All that said, this library was written by a developer who hasn't been with the company for a few years, so if there is a larger refactor that you think would make things cleaner, we can make time for that.

Reply:
Thanks @patrickomalley. I will go ahead and use fatalError() in this case. 

I think that this type of errors, i.e. programming errors, should be made visible as soon as possible so we can catch them in dev stages. We should establish a clear a way of treating errors so we don't second guess and we are consistent across all projects. 

Something along the lines of:

**Programming errors should crash early.** 
If we are making a programming error, like accessing an array index that is out of bounds, the program should crash ( like it does if we actually try to do that). 

Examples: 
- xib file, or image, or font is missing from the bundle. 
- segue identifier is wrong
- storyboard doesn't contain an initial view controller
- json config file contains wrong or incomplete entities
- core data model is incorrect
- etc.

**Data errors should throw an exception**
If, on the other hand, the error comes from the data, we should handle that error so the program doesn't crash.

Examples:
- Anything coming from a backend response that has the incorrect format, or incorrect data, like a missing field for example.
- Anything coming from data produced by the user, like some ID that should have certain format and it was not validated properly by the UI. 
- Etc